h2/src/main/org/h2/command/Parser.java

```diff
@@ -7528,6 +7528,11 @@ private Prepared parseSet() {
             readIfEqualOrTo();
             read();
             return new NoOperation(session);
+        } else if (readIf("STATEMENT_TIMEOUT")){
+            // for PostgreSQL compatibility
+            readIfEqualOrTo();
+            read();
+            return new NoOperation(session);
```

Comment:
I think it should create a `Set` operation with `SetTypes.QUERY_TIMEOUT` type.

Please, also add a test case to `TestCompatibility`, for example.

Reply:
Can I add a function like `SLEEP()` (for MySQL) or `PG_SLEEP()` (for PostgreSQL) to test query timeout? I'm unable to find out such a statement to delay seconds in H2.

Otherwise, I have to open 2 connections to simulate timeout: one connection to lock a row, sleep from client and free lock; another to get lock and wait for timeout.

Reply:
You can use `Statement.getQueryTimeout()` (beware: it caches the result, don't use it twice), or `SELECT VALUE FROM INFORMATION_SCHEMA.SETTINGS WHERE NAME = 'QUERY_TIMEOUT'` to check that value was really set.

Time units are different. `Statement.getQueryTimeout()` returns seconds, H2's own `QUERY_TIMEOUT` uses milliseconds.